### ISSUE TITLE
Make undefined Particle attributes return NaNs instead of raising exceptions

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#1016]
Right now, if we try to access an undefined `Particle` attribute, an exception is raised.
```Python
>>> tau_neutrino = Particle("tau neutrino")
>>> tau_neutrino.mass
MissingAtomicDataError: the mass of nu_tau is not available.
```
The problem with this is that if we're looping over a bunch of particles, we could end up having to deal with the exception as a special case.  I propose that instead of raising an exception for situations like this, we return `numpy.nan`, multiplied by the expected units when appropriate.  We can then let `numpy.nan` take care of keeping track of whether it is defined or not.

Fixing this issue would make #948 a little bit easier to address.  

Returning `numpy.nan` in the appropriate units would also make the behavior of `Particle` and `CustomParticle` more consistent.

**Comment by StanczakDominik** [comment]
One minor worry about this scheme that I have: multiple sources of errors (which can be handled by separate except clauses) would become a single np.nan, which is less informative and cannot be handled on a case by case basis. I

I do like the direction, though!

**Comment by namurphy** [comment]
> One minor worry about this scheme that I have: multiple sources of errors (which can be handled by separate except clauses) would become a single np.nan, which is less informative and cannot be handled on a case by case basis.

Agreed — I share that worry.  One possibility would be to limit this to attributes which are appropriate for a category of particle, but which currently raise a `MissingParticleDataError`.  For example, `mass` and `mass_energy` for neutrinos would return `np.nan * u.whatever`, but `standard_atomic_weight` would raise an exception saying that this parameter is not defined for a neutrinos.  

(Creating a particle hierarchy as proposed by #687 would let us avoid having attributes like `standard_atomic_weight` for things that are not atoms, but that's beyond the scope of this issue.)

We could still issue warnings about missing particle data, though.

**Comment by alfkoehn** [comment]
I just wanted to look into this, but I'm a bit lost here... just wondering where the information about e.g. the particle's mass is stored (is there a dictionary somewhere which I missed or have I missed something completely different) ?

**Comment by ChrisMoriarty** [comment]
I'm just posting to say I'm planning to work on this issue as part of the Central Engineering "Engage an Engineer" effort. 

From our kickoff meeting with Nick, the idea was for me to pick a "Good first contribution" issue to use to review the Documentation for contributing, and also to become a little more familiar with the code with intention to start helping with code reviews.

**Comment by Sobeskes** [comment]
Hey, I saw that this issue hasn't seen much activity lately, I hope you all wouldn't mind if I tried taking it on.

**Comment by ChrisMoriarty** [comment]
I don't mind! Sorry for letting it sit for so long.

**Comment by Sobeskes** [comment]
This was resolved by #1825, but due to a typo in the PR I didn't mention it there until now.

**Comment by StanczakDominik** [comment]
Good point! Let's close this then, thank you! :)
